### DISCUSSION TITLE
How to enable Login button within the Landing Pages instead of whole site

### DISCUSSION THREAD

**Discussion by mohmmadali1976** [oidc-wp/openid-connect-generic#551] [Q&A]
Hi,

First of all, I would like to thank the whole team for this awesome plugin that supports SSO. Currently, we are building portals using WordPress and have enabled authentication against an identity server using the OpenID Connect Generic plugin. It's working perfectly with the current configuration.

We would like to extend our portal's functionality further. Initially, I want to have a landing page (static content) with an OpenID Connect login button shortcode, and I want the authentication flow to start from there onwards. However, what I see is the default WordPress login page with the login button appended to it, rather than showing the landing page with the login button.

It appears that this plugin applies its functionality throughout the WordPress site rather than on a specific page. Is there a way to customize this to apply only to a specific page rather than the entire site? Additionally, when a user logs out, I would like to show a custom landing page or static page instead of the default WordPress login.

Your assistance on this matter would be greatly appreciated.

Best Regards,
![image](https://github.com/oidc-wp/openid-connect-generic/assets/69623854/01b4a6c2-b07e-409f-8b81-efcf70d9b05c)

Ali

**Comment by timnolte** [comment]
There is a button shortcode, as well as a URL shortcode for adding to your own button design, available by the plugin. You should be able add these any page you like on a site.

You will find details here: https://github.com/oidc-wp/openid-connect-generic/wiki/Shortcodes

**Reply by mohmmadali1976** [reply]
Appreciate your prompt response. I have reviewed the shortcodes but still have a couple of queries. Please bear with me.

Here are the steps I've taken so far:

1. Activated the plugin.
2. Input the IDP details.
3. Now my site is ready to incorporate a Login button or Login URL as per the documentation, which is working perfectly.

In the Plugin settings, we have an option:
- **Enforce Privacy:** Requires users to be logged in to see the site. (This setting is crucial for handling Portals.)

Let's consider my site URL is https://example.com/home, which is static content with a Login Button—this serves as the entry point for clients. 

If I select "Enforce Privacy," attempting to access https://example.com/home will redirect to the default WordPress login page. Alternatively, if not enforced, accessing https://example.com/home allows us to initiate the flow by clicking on the Login Button.

In the latter scenario, pages like https://example.com/dashboard, https://example.com/customers, https://example.com/notes, which are supposed to be accessible post-login, will also be accessible without requiring a login.

Therefore, my question is, even if I select "Enforce Privacy," is there a way to display the https://example.com/home page instead of redirecting to the default login page?

Thank you for your assistance.

Best regards

**Reply by timnolte** [reply]
So, by default this plugin is really designed for securing your WordPress Dashboard and is not primarily intended to be used as more of a "membership" style site. While yes the Enforce Privacy mode is to a degree designed with securing the entire site it is in fact meant to still rely on the standard core WordPress login process without providing any "public" facing content. You could enable the "auto" mode instead of the "button" mode which would reduce exposing people to the standard WordPress login screen, but that would also prevent providing some sort of "public" on-boarding content.

I think we'd have to consider adding the ability to configure excluded URLs/pages that would bypass the need for authentication to allow what you are trying to do.

**Reply by mohmmadali1976** [reply]
Yes, "auto" mode is perfectly working if we have two different sites.

https://example.com -> [Static Website] -> From here I can redirect to https://test.com (OpenID Connect Generic Plugin activated with Auto mode) : I tried this and perfect solution.

However as per your suggestion 

excluded URLs/pages, is it possible today? or any need to extend plugin capability further?

**Comment by mohmmadali1976** [comment]
Hi,
Could you please provide the solution? I appreciate your help as I am currently stuck here.
Best regards,

**Reply by timnolte** [reply]
There is not a solution without code changes to the plugin.

**Comment by mohmmadali1976** [comment]
Okay, do you have any plans to include this feature? Because in the issues, I have come across this:

Exclude pages on "enforce_privacy" sites #494

And changing the code will have an impact on Updates
